src/main/java/kitchenpos/shared/domain/DomainEntity.java

```diff
@@ -0,0 +1,37 @@
+package kitchenpos.shared.domain;
+
+import java.io.Serializable;
+
+public abstract class DomainEntity<T extends DomainEntity<T, TID>, TID> implements Serializable {
+    @Override
+    public boolean equals(Object other) {
+        if (other == null) {
+            return false;
+        }
+
+        return equals((T)other);
+    }
+
+    public boolean equals(T other) {
+        if (other == null) {
+            return false;
+        }
+
+        if (getId() == null) {
+            return false;
+        }
```

Comment:
사소하지만 아래처럼 변경해볼 수도 있을 것 같네요. 

```suggestion
        if (other == null || getId() == null) {
            return false;
        }
```